client/extensions/zoninator/state/zones/reducer.js

```diff
@@ -17,6 +23,17 @@ export const requesting = createReducer( {}, {
 
 export const items = createReducer( {}, {
 	[ ZONINATOR_UPDATE_ZONES ]: ( state, { siteId, data } ) => ( { ...state, [ siteId ]: data } ),
+	[ ZONINATOR_UPDATE_ZONE ]: ( state, { siteId, data } ) => {
+		const zones = [ ...( state[ siteId ] || [] ) ];
+		const idx = findIndex( zones, { term_id: data.term_id || -1 } );
```

Comment:
I don't think you need the `|| -1` here since `findIndex` will return that if it doesn't find the element.